src/main/java/reposense/system/LogsManager.java

```diff
@@ -27,6 +31,7 @@ public class LogsManager {
     // eg. reposense.log.0, in the logs/ folder of the working directory
     private static final String LOG_FOLDER_NAME = "logs";
     private static final String LOG_FILE_NAME = "reposense.log";
+    private static final Path TEMP_LOG_DIR = Paths.get(System.getProperty("user.dir"));
```

Comment:
I'm not sure if this fixes the concurrency issue mentioned in #1908/#1856. According to the issue description, the problem seems to be coming from the fact that LogsManager is a static variable, which happened to be accessed at the same time.

At this point, I don't seem to recall having seen the problem as mentioned in #1856 which was closed and resolved. I believe that issue was primarily due to unsynchronized access to the LogsManager. I might be missing something though so please let me know otherwise!

Reply:
To clarify, it's not the LogsManager class itself that causes the concurrency issue but the list of loggers inside LogsManager where all their output folders have to be changed every time the `ArgsParser::parse` is called (see the `synchronized` block in `LogsManager`). On a normal run, the method is only called once but  in tests, it's called multiple times. So the list has a chance of being accessed concurrently which causes the issue. 

This fixes it in a roundabout way by only changing a single variable which could on paper lead to another type of concurrency issue (multiple writes to a single variable) that I believe is not caught but it's also not one that matters in this context since we just have to avoid the concurrency exception being thrown since the actual logs output folder does not affect our test cases.